spec-tests/src/test/java/org/candlepin/spec/owners/OwnerResourceOwnerInfoSpecTest.java

```diff
@@ -0,0 +1,135 @@
+        throws ApiException {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = createOwner(adminClient);
+        ApiClient userClient = createUserClient(UserUtil.createUser(adminClient, owner));
+
+        ProductDTO tmpProduct = Products.withAttributes(
+            ProductAttributes.Version.withValue("6.4"),
+            ProductAttributes.Sockets.withValue("2"),
+            ProductAttributes.MultiEntitlement.withValue("true")
+        );
+        ProductDTO product = adminClient.ownerProducts().createProductByOwner(owner.getKey(), tmpProduct);
+        adminClient.owners().createPool(owner.getKey(), Pools.random(product));
+
+        createConsumer(userClient, Consumers.random(owner));
+        createConsumer(userClient, Consumers.random(owner).facts(Map.of("virt.is_guest", "true")));
+
+        ApiClient userClientWithPerms = createUserClient(
+            UserUtil.createWith(adminClient, Permissions.USERNAME_CONSUMERS.readOnly(owner)));
+
+        PoolDTO pool = adminClient.owners().listOwnerPools(owner.getKey(), null, null,
+            product.getId(), null, null, null, null, null,
+            null, null, null, null, null, null, null, null).get(0);
+        assertNotNull(pool);
+
+        // Create a physical system with valid status
+        ConsumerDTO consumer1 = createConsumer(
+            userClientWithPerms, Consumers.random(owner, ConsumerTypes.System)
+            .facts(Map.of("cpu.cpu_socket(s)", "4"))
+            .installedProducts(Set.of(createInstalledProduct(product.getId(), product.getName()))));
+        ApiClient consumerClient1 = ApiClients.ssl(consumer1);
+        JsonNode ents = consumerClient1.consumers().bindPool(consumer1.getUuid(), pool.getId(), 1);
+        assertNotNull(ents);
+
+        // Create a guest system with a partial status
+        ConsumerDTO consumer2 = createConsumer(
+            userClientWithPerms, Consumers.random(owner, ConsumerTypes.System)
+            .facts(Map.of("virt.is_guest", "true"))
+            .installedProducts(Set.of(createInstalledProduct(product.getId(), product.getName()))));
+
+        ApiClient consumerClient2 = ApiClients.ssl(consumer2);
+        ents = consumerClient2.consumers().bindPool(consumer2.getUuid(), pool.getId(), 1);
+        assertNotNull(ents);
+
+        // Create a guest system with invalid status
+        createConsumer(userClientWithPerms, Consumers.random(owner, ConsumerTypes.System)
+            .facts(Map.of("virt.is_guest", "true"))
+            .installedProducts(Set.of(createInstalledProduct(product.getId(), product.getName()))));
+
+        OwnerInfo info = userClient.owners().getOwnerInfo(owner.getKey());
+        assertEquals(5, info.getConsumerCounts().get("system").intValue());
+
+        OwnerInfo systemOwnerInfo = userClientWithPerms.owners().getOwnerInfo(owner.getKey());
+        assertEquals(3, systemOwnerInfo.getConsumerCounts().get("system").intValue());
```

Comment:
These would be easier to read as map assertions:

```
assertThat(systemOwnerInfo.getConsumerCounts())
  .isNotNull()
  .containsEntry("system", 3);

assertThat(systemOwnerInfo.getConsumerGuestCounts())
  .isNotNull()
  .containsEntry("physical", 1)
  .containsEntry("guest", 2);

...
```

Reply:
done